### PR TITLE
SQL: TRX_ID to TIMESTAMP versioning switch [fixes #428]

### LINKED ISSUE #428: TRX_ID to TIMESTAMP versioning switch
### Reproduce
```sql
create table t (
  a int,
  b int,
  s bigint unsigned as row start,
  e bigint unsigned as row end,
  period for system_time(s, e)
) engine=innodb with system versioning;

insert into t values (1,1);
select * from t for system_time all;

alter table t drop column s, drop column e;
select * from t for system_time all;
```

### Result
```sql
create table t (
a int,
b int,
s bigint unsigned as row start invisible,
e bigint unsigned as row end invisible,
period for system_time(s, e)
) engine=innodb with system versioning;
insert into t values (1,1);
select * from t for system_time all;
a	b	s	e
1	1	1284	18446744073709551615
alter table t drop column s, drop column e;
select * from t for system_time all;
a	b	s	e
1	1	2017-12-23 20:45:15.010647	2038-01-19 06:14:07.999999
```

### Expected
Versioning must to switch from trx_id-based to timestamp-based.

### PR DESCRIPTION
**Pull request by kevgs**

### CODE REVIEW

File: mysql-test/suite/versioning/r/trx_id.result [hunk 1]
```diff
@@ -79,6 +79,18 @@ algorithm=copy;
...
+# TRX_ID to TIMESTAMP versioning switch
+create or replace table t1 (
+x int,
+s bigint unsigned as row start invisible,
+e bigint unsigned as row end invisible,
+period for system_time (s, e)
+) engine innodb with system versioning;
+insert into t1 values (1);
+alter table t1 drop column s, drop column e;
+select e from t1 for system_time all;
```
Comment (midenok) [inline mysql-test/suite/versioning/r/trx_id.result]:
```sql
select sys_end = 18446744073709551615 as transaction_based from t1;
```

File: mysql-test/suite/versioning/r/trx_id.result [hunk 2]
```diff
@@ -79,6 +79,18 @@ algorithm=copy;
 select check_result(count(*) = @tmp) from mysql.transaction_registry;
 check_result(count(*) = @tmp)
 [CORRECT]
+# TRX_ID to TIMESTAMP versioning switch
+create or replace table t1 (
+x int,
+s bigint unsigned as row start invisible,
+e bigint unsigned as row end invisible,
```
Comment (midenok) [inline mysql-test/suite/versioning/r/trx_id.result]: sys_end

File: mysql-test/suite/versioning/r/trx_id.result [hunk 3]
```diff
@@ -79,6 +79,18 @@ algorithm=copy;
 select check_result(count(*) = @tmp) from mysql.transaction_registry;
 check_result(count(*) = @tmp)
 [CORRECT]
+# TRX_ID to TIMESTAMP versioning switch
+create or replace table t1 (
+x int,
+s bigint unsigned as row start invisible,
```
Comment (midenok) [inline mysql-test/suite/versioning/r/trx_id.result]: sys_start

File: sql/handler.cc [hunk 4]
```diff
@@ -7369,8 +7376,9 @@ bool Vers_parse_info::fix_alter_info(THD *thd, Alter_info *alter_info,
           return true;
         }
 
-        if (vers_change_sys_field(thd, name, alter_info,
-                                  f->flags & VERS_SYSTEM_FIELD, name))
+        bool integer= table->vers_start_field()->type() == MYSQL_TYPE_LONGLONG;
+        if (vers_change_sys_field(thd, name, alter_info, f->flags & VERS_SYSTEM_FIELD, name,
+                                  integer))
```
Comment (midenok) [inline sql/handler.cc]: Remove `vers_change_sys_field()`.

Comment (kevgs) [reply sql/handler.cc]: Are you sure? That will change UI. `DROP sys_start, DROP sys_end` is currently required step before `DROP SYSTEM VERSIONING` as requested by @vuvova. I'm gonna ask him about this anyways but this PR is only about buggy versioning switch.

Comment (midenok) [reply sql/handler.cc]: I mean unwrap code from `vers_change_sys_field()`. The function is redundant, too much args, some duplicated.

Comment (kevgs) [reply sql/handler.cc]: Done.

File: sql/handler.cc [hunk 5]
```diff
@@ -7369,11 +7360,16 @@ bool Vers_parse_info::fix_alter_info(THD *thd, Alter_info *alter_info,
...
-                                  f->flags & VERS_SYSTEM_FIELD, name))
-        {
+        bool integer= table->vers_start_field()->type() == MYSQL_TYPE_LONGLONG;
+        Create_field *field= vers_init_sys_field(thd, name, f->flags & VERS_SYSTEM_FIELD, integer);
+        if (!field)
           return true;
-        }
+
+        field->change.str= name;
+        field->change.length= strlen(name);
```
Comment (midenok) [inline sql/handler.cc]: Did you check that `strlen()` of **name** is known?

Comment (kevgs) [reply sql/handler.cc]: That can be improved. Thanks for noticing.

Comment (kevgs) [reply sql/handler.cc]: Fixed.

**Review by midenok — APPROVED** [review]

**Pull request merged**